packages/grid/_modules_/grid/hooks/features/rows/useGridEditRows.ts

```diff
@@ -171,13 +171,15 @@ export function useGridEditRows(apiRef: GridApiRef) {
 
   const setCellValue = React.useCallback(
```

Comment:
Shouldn't `setEditCellProps` be handled too. For instance, https://deploy-preview-1785--material-ui-x.netlify.app/components/data-grid/editing/#controlled-editing

---

`Age` is supposed to be a number but is stored as a string. Shouldn't we update all the default columns to leverage this new `valueParser` property?

<img width="454" alt="Capture d’écran 2021-05-30 à 21 41 13" src="https://user-images.githubusercontent.com/3165635/120117891-cc7a8e00-c18f-11eb-9bf1-e10306a6d0e1.png">

Reply:
> Shouldn't setEditCellProps be handled too. For instance, https://deploy-preview-1785--material-ui-x.netlify.app/components/data-grid/editing/#controlled-editing

I think we need to discuss this. From what I understood, `setEditCellProps` should store in the `editRows` state the new value for the edit cell, which feeds the `value` prop. The value parser would only be applied when updating the row with the value stored in the state (a.k.a commiting). By the way, we use different approaches to create edit components:
- `GridEditInputCell` and `GridEditBooleanCell` publish the `GRID_CELL_EDIT_PROPS_CHANGE` event
- The `ValidateServerNameGrid` [demo](https://deploy-preview-1785--material-ui-x.netlify.app/components/data-grid/editing/#client-side-validation) calls `setEditCellProps`
- The [demo](https://deploy-preview-1785--material-ui-x.netlify.app/components/data-grid/editing/#custom-edit-component) with the rating component calls `commitCellChange`.

> Age is supposed to be a number but is stored as a string. Shouldn't we update all the default columns to leverage this new valueParser property?

We could, but what about update `GridEditInputCell` to parse the value and convert it to a number? We already do something similar with dates.

Reply:
@m4theushw I don't share the same perspective. Take this example:

https://github.com/mui-org/material-ui-x/blob/a24eba2a67657afe2a608bd29adc52fd629a7ca2/packages/grid/_modules_/grid/components/cell/GridEditInputCell.tsx#L41

We are doing the formatting manually inside the field for each value change.
For the filtering, we are also applying the formatted manually: https://github.com/mui-org/material-ui-x/blob/a24eba2a67657afe2a608bd29adc52fd629a7ca2/packages/grid/_modules_/grid/models/colDef/gridDateOperators.ts#L23

IMHO, we could abstract both with the new `valueParser` feature (we currently have two different implementations of `parseDate` and `parseDateTime`).

> We might ended up parsing an already parsed value if we choose to call valueParser in all places.

Then this sounds like we would uncover root issues with the data flow, that makes wrong assumptions.